### ISSUE TITLE
ddev-router does not start with typo project

### ISSUE THREAD

**Issue by manuschillerdev** [drud/ddev#1890]
**Describe the bug**

I can not spin up a typo3 project as described in https://speakerdeck.com/peterkraume/local-development-environment-for-typo3-with-ddev?slide=7 since i get the following error:

```
Failed to start app typo to run create-project: ddev-router failed to become ready: logOutput=2019/10/21 07:15:53 [emerg] 1639#1639: no servers are inside upstream in /etc/nginx/conf.d/default.conf:89
nginx: [emerg] no servers are inside upstream in /etc/nginx/conf.d/default.conf:89
nginx: configuration file /etc/nginx/nginx.conf test failed
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
curl: (22) The requested URL returned error: 404 Not Found
ddev-router healthcheck endpoint not responding
, err=container /ddev-router unhealthy: 2019/10/21 07:15:53 [emerg] 1639#1639: no servers are inside upstream in /etc/nginx/conf.d/default.conf:89
nginx: [emerg] no servers are inside upstream in /etc/nginx/conf.d/default.conf:89
nginx: configuration file /etc/nginx/nginx.conf test failed
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
curl: (22) The requested URL returned error: 404 Not Found
ddev-router healthcheck endpoint not responding
```

**To Reproduce**
1. `brew tap drud/ddev`
2. `brew install ddev`
3. `mkdir typo && cd typo`
4. `ddev config --project-type php`
5. `ddev composer create typo3/cms-base-distribution ^9 --no-interaction --prefer-dist`
6. `ddev start`

=> the described error occurs

**Expected behavior**
I can start up a typo3 instance as describe in the presentation without `ddev-router` issues

**Version and configuration information (please complete the following information):**
 - Host computer Macbook Pro late 2017, Mojave 10.14.6
 - Docker version information:
```
Client: Docker Engine - Community
 Version:           19.03.2
 API version:       1.40
 Go version:        go1.12.8
 Git commit:        6a30dfc
 Built:             Thu Aug 29 05:26:49 2019
 OS/Arch:           darwin/amd64
 Experimental:      false

Server: Docker Engine - Community
 Engine:
  Version:          19.03.2
  API version:      1.40 (minimum version 1.12)
  Go version:       go1.12.8
  Git commit:       6a30dfc
  Built:            Thu Aug 29 05:32:21 2019
  OS/Arch:          linux/amd64
  Experimental:     true
 containerd:
  Version:          v1.2.6
  GitCommit:        894b81a4b802e4eb2a91d1ce216b8817763c29fb
 runc:
  Version:          1.0.0-rc8
  GitCommit:        425e105d5a03fabd737a126ad93d62a9eeede87f
 docker-init:
  Version:          0.18.0
  GitCommit:        fec3683
```
 - ddev version information (use `ddev version`)
```
DDEV-Local version	v1.11.2
commit            	v1.11.2
db                	drud/ddev-dbserver:v1.11.0-10.2
dba               	drud/phpmyadmin:v1.11.0
ddev-ssh-agent    	drud/ddev-ssh-agent:v1.11.0
docker            	19.03.2
docker-compose    	1.24.1
os                	darwin
router            	drud/ddev-router:v1.11.0
web               	drud/ddev-webserver:v1.11.0
```
 - config.yaml contents for the misbehaving project
used defaults
 - Do you have any custom configuration (nginx, php, mysql) in the .ddev folder? If so, have you tried without them?
no

**Comment by rfay** [comment]
Wow, that's surprising, I do that regularly. Please do a `ddev poweroff` and then another `ddev start`. 

What version of docker do you have? `docker --version`

**Comment by manuschillerdev** [comment]
@rfay thanks for the quick response.
Docker: `Docker version 19.03.2, build 6a30dfc`

**Comment by manuschillerdev** [comment]
i also can not delete any "broken" projects: 

`ddev stop --remove-data typo3`
=> 
```
Must start non-running project to do database snapshot
Building ddev-ssh-agent
Creating ddev-ssh-agent ... done

ssh-agent container is running: If you want to add authentication to the ssh-agent container, run 'ddev auth ssh' to enable your keys.
Pushing mkcert rootca certs to ddev-global-cache
Pushed mkcert rootca certs to ddev-global-cache
Failed to stop project typo3:
Failed to start project to perform database snapshot
```

**Comment by rfay** [comment]
Please use `ddev stop --remove-data --omit-snapshot <project1> <project2>` to get rid of those, then do a `ddev poweroff`, then try your start again.

I think you might have had duplicate projects with the same project name, and that will definitely blow up the router.

**Comment by manuschillerdev** [comment]
the ddev-router works now. 

but `ddev composer create typo3/cms-base-distribution ^9 --no-interaction --prefer-dist` fails with another error now:

<img width="679" alt="Bildschirmfoto 2019-10-21 um 10 41 50" src="https://user-images.githubusercontent.com/56154253/67190018-7f4d0c80-f3ef-11e9-9733-13e4ee107e3d.png">

**Comment by rfay** [comment]
Please use `ddev composer create "typo3/cms-base-distribution:^9" --no-interaction --prefer-dist
` as suggested in https://ddev.readthedocs.io/en/latest/users/cli-usage/#typo3-quickstart

**Comment by manuschillerdev** [comment]
thanks to @rfay I found out what the problem was:

The slides I used as source:
https://speakerdeck.com/peterkraume/local-development-environment-for-typo3-with-ddev?slide=7
and
https://de.slideshare.net/certifuncation/2019certifuncationddevfortypo3 (2019)

... suggest to specify the typo version as 
`ddev composer create typo3/cms-base-distribution ^9 --no-interaction --prefer-dist` which uses the target version `^9` as typo3 project name and therefore created a corrupted folder structure on  my machine.

changing this command to `ddev composer create "typo3/cms-base-distribution:^9" --no-interaction --prefer-dist` and sticking to the tutorial at https://ddev.readthedocs.io/en/latest/users/cli-usage/#typo3-quickstart as suggested by @rfay solves all my problems.

ddev / typo works like a charm now, thanks! @rfay
